notes/klustered-teams/episode-3/skyscanner/README.md

```diff
@@ -0,0 +1,33 @@
+# Kluster 000
+
+## Discovered Symptoms
+
+
+## Contributing Factors
+
+
+## Notes from Kluster Breaker
+
+### APIServer Break
+
+We modified the core `v1.` [apiservice](https://kubernetes.io/docs/tasks/extend-kubernetes/configure-aggregation-layer/#register-apiservice-objects) object to point to the `kubernetes` service in the `default` namespace, resulting in the APIServer being unable to ever successfully break out of a loop and respond as all API requests got trapped in a loop. The reason restarting the APIServer was enough to fix this was that the built-in aggregator server which manages aggregation of APIs within kubernetes rebuilds this list on startup from the API resources defined within the Kube-APIServer.
+
+### Cilium/Networking Break
+
+Cilium is able to replace [kube-proxy](https://kubernetes.io/docs/concepts/overview/components/#kube-proxy) when configured correctly, taking over the maintenance of IPTable rules to allow services within the cluster to be successfully reached. Here we both updated the Cilium config to disable this behaviour, without installing kube-proxy to take over, but also removed the lifecycle related healthchecks from the Cilium Daemonset to enable the new pods to launch successfully and appear to be healthy.
+
+One of the difficult things here is just how many options there are for configuring Cilium due to its power.
+
+### ValidatingAdmissionWebhook
+
+Nothing clever here, we just created a new [ValidatingAdmissionWebhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/) for the creation or update of pods which pointed at a random URL and had its `failurePolicy` set to fail. This would have blocked the creation of new pods as the klustered deployment was updated if it hadn't already been deleted.
```

Comment:
To be fair, we didn't specify a `failurePolicy` because by default is set to fail.

Reply:
If we want the yaml

```yaml
apiVersion: admissionregistration.k8s.io/v1
kind: ValidatingWebhookConfiguration
metadata:
  name: default.k8s.io
webhooks:
- name: default.k8s.io
  rules:
  - apiGroups:   [""]
    apiVersions: ["v1"]
    operations:  ["CREATE", "UPDATE"]
    resources:   ["pods"]
    scope:       "Namespaced"
  clientConfig:
    url: https://www.google.com
  admissionReviewVersions: ["v1", "v1beta1"]
  sideEffects: None
  timeoutSeconds: 5
 ```

Reply:
Nice to point out that `https://www.google.com` was returning error 413 that might have been a little too misleading.

Reply:
Added, and updated, as appropriate.